x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx

```diff
@@ -0,0 +1,327 @@
+  dataRetention,
+  dataStreamName,
+  configuredByILM,
+  onClose,
+}) => {
+  const { size, unit } = splitSizeAndUnits(dataRetention);
+  const {
+    services: { notificationService },
+  } = useAppContext();
+
+  const { form } = useForm({
+    defaultValue: {
+      dataRetention: size,
+      timeUnit: unit || 'd',
+      infiniteRetentionPeriod: !dataRetention,
+    },
+    schema: configurationFormSchema,
+    id: 'editDataRetentionForm',
+  });
+  const [formData] = useFormData({ form });
+  const isDirty = useFormIsModified({ form });
+
+  const onSubmitForm = async () => {
+    const { isValid, data } = await form.submit();
+
+    if (!isValid) {
+      return;
+    }
+
+    return updateDataRetention(dataStreamName, data).then(({ data: responseData, error }) => {
+      if (responseData) {
+        const successMessage = i18n.translate(
+          'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.successDataRetentionNotification',
+          { defaultMessage: 'Data retention updated' }
+        );
+        notificationService.showSuccessToast(successMessage);
+
+        return onClose({ hasUpdatedDataRetention: true });
+      }
+
+      if (error) {
+        const errorMessage = i18n.translate(
+          'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.errorDataRetentionNotification',
+          {
+            defaultMessage: "Error updating data retention: '{error}'",
+            values: { error: error.message },
+          }
+        );
+        notificationService.showDangerToast(errorMessage);
+      }
+
+      onClose();
+    });
+  };
+
+  return (
+    <EuiModal onClose={() => onClose()} data-test-subj="editDataRetentionModal">
+      <Form form={form} data-test-subj="editDataRetentionForm">
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage
+              id="xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.modalTitleText"
+              defaultMessage="Edit data retention"
+            />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          {configuredByILM && (
+            <>
+              <EuiCallOut
+                color="warning"
+                iconType="warning"
+                data-test-subj="configuredByILMWarning"
+                title={i18n.translate(
+                  'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.configuredByILMWarningTitle',
+                  {
+                    defaultMessage: 'Override ILM data retention',
+                  }
+                )}
+              >
+                <p>
+                  <FormattedMessage
+                    id="xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.configuredByILMWarningText"
+                    defaultMessage="Since this data stream is configured by ILM, you must also update the ILM policy to match the new data retention period."
```

Comment:
```suggestion
                    defaultMessage="The data retention set for this data stream currently overrides the retention settings of the Index Lifecycle Policy attached to this data stream. Update the corresponding policy to match the retention period you need."
```

I'm not clear on the behavior implied by this message, so not sure what to suggest here for the title and description. So feel free to ignore or correct my suggestion here.
Do settings conflict? What happens or which setting counts if values are different here and in the policy?

Reply:
I think we might leave this as is now, as we are going to update it in the upcoming days to address a bunch of scenarios that could happen here. See https://github.com/elastic/kibana/issues/167414

Reply:
Sounds good, let me know when I can help :)